src/Thinktecture.Relay.Server/Middleware/RelayMiddleware.cs

```diff
@@ -314,4 +323,31 @@ private async Task<bool> TryInlineBodyContentAsync(TRequest request, Cancellatio
 		CancellationToken cancellationToken)
 		=> _responseWriter.WriteAsync(_relayContext.ClientRequest.CreateResponse<TResponse>(httpStatusCode), response,
 			cancellationToken);
+
+	private async Task<Guid?> LoadTenantByIdAsync(string name, CancellationToken cancellationToken = default)
+	{
+		var normalizedName = _tenantService.NormalizeName(name);
+		var cacheKey = $"{normalizedName}_id";
+
+		var cachedId = await _cache.GetAsync(cacheKey, cancellationToken);
+		if (cachedId != null)
+		{
+			return new Guid(cachedId);
+		}
+
+		var tenant = await _tenantService.LoadTenantByNameAsync(name);
+		if (tenant == null)
+		{
+			return null;
```

Comment:
We need to cache the negatives, too. Maybe a two-step mechanism? First check a single entry if the name is unknown, then if the name is already known.